src/Umbraco.Examine/MediaValueSetBuilder.cs

```diff
@@ -29,6 +34,28 @@ public override IEnumerable<ValueSet> GetValueSets(params IMedia[] media)
             foreach (var m in media)
             {
                 var urlValue = m.GetUrlSegment(_urlSegmentProviders);
+
+                var umbracoFilePath = string.Empty;
+                var umbracoFile = string.Empty;
+
+                var umbracoFileSource  = m.GetValue<string>(Constants.Conventions.Media.File);
+
+                if (umbracoFileSource.DetectIsJson())
+                {
+                    var cropper = JsonConvert.DeserializeObject<ImageCropperValue>(m.GetValue<string>(Constants.Conventions.Media.File));
+                    if (cropper != null)
+                    {
+                        umbracoFilePath = cropper.Src;
+                    }
+                }
+                else
+                {
+                    umbracoFilePath = umbracoFileSource;
+                }
+
+                var uri = new Uri(_runtimeState.ApplicationUrl.GetLeftPart(UriPartial.Authority) + umbracoFilePath);
```

Comment:
Null or empty checks are needed for umbracoFilePath before using it. And then below when adding to the index, it should use the same notation as the other fields and use the Yield ext method else insert an empty array not an array with an empty element

Reply:
Add null check and Yield